logstash-core/lib/logstash/compiler.rb

```diff
@@ -19,31 +19,14 @@
 
 java_import org.logstash.config.ir.PipelineIR
 java_import org.logstash.config.ir.graph.Graph
+java_import org.logstash.config.ir.ConfigCompiler
 
 module LogStash; class Compiler
   include ::LogStash::Util::Loggable
 
-  def self.compile_sources(sources_with_metadata, support_escapes)
-    graph_sections = sources_with_metadata.map do |swm|
-      self.compile_graph(swm, support_escapes)
-    end
-
-    input_graph = Graph.combine(*graph_sections.map {|s| s[:input] }).graph
-    output_graph = Graph.combine(*graph_sections.map {|s| s[:output] }).graph
-
-    filter_graph = graph_sections.reduce(nil) do |acc, s|
-      filter_section = s[:filter]
-
-      if acc.nil?
-        filter_section
-      else
-        acc.chain(filter_section)
-      end
-    end
-
-    original_source = sources_with_metadata.map(&:text).join("\n")
-
-    PipelineIR.new(input_graph, filter_graph, output_graph, original_source)
+  # Used only in lir_serializer_spec
```

Comment:
if its only used in specs, maybe move `ConfigCompiler.compileSources(source_with_metadata, support_escapes)`
into a spec helper/method there? ... (than we probably could get rid of most java_imports), wdyt?

Reply:
Sounds good